drivers/gpio/gpio_renesas_rz.c

```diff
@@ -0,0 +1,496 @@
+#include "gpio_renesas_rz.h"
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(rz_gpio, CONFIG_GPIO_LOG_LEVEL);
+
+#define LOG_DEV_ERR(dev, format, ...) LOG_ERR("%s:" #format, (dev)->name, ##__VA_ARGS__)
+#define LOG_DEV_DBG(dev, format, ...) LOG_DBG("%s:" #format, (dev)->name, ##__VA_ARGS__)
+
+struct gpio_rz_config {
+	struct gpio_driver_config common;
+	uint8_t ngpios;
+	uint8_t port_num;
+	bsp_io_port_t fsp_port;
+	const ioport_cfg_t *fsp_cfg;
+	const ioport_api_t *fsp_api;
+	const struct device *int_dev;
+	uint8_t tint_num[GPIO_RZ_MAX_TINT_NUM];
+};
+
+struct gpio_rz_data {
+	struct gpio_driver_data common;
+	sys_slist_t cb;
+	ioport_instance_ctrl_t *fsp_ctrl;
+	struct k_spinlock lock;
+};
+
+struct gpio_rz_tint_isr_data {
+	const struct device *gpio_dev;
+	gpio_pin_t pin;
+};
+
+struct gpio_rz_tint_data {
+	struct gpio_rz_tint_isr_data tint_data[GPIO_RZ_MAX_TINT_NUM];
+	uint32_t irq_set_edge;
+};
+
+struct gpio_rz_tint_config {
+	void (*gpio_int_init)(void);
+};
+
+static int gpio_rz_pin_config_get_raw(bsp_io_port_pin_t port_pin, uint32_t *flags);
+
+#ifdef CONFIG_GPIO_GET_CONFIG
+static int gpio_rz_pin_get_config(const struct device *dev, gpio_pin_t pin, gpio_flags_t *flags)
+{
+	const struct gpio_rz_config *config = dev->config;
+	bsp_io_port_pin_t port_pin = config->fsp_port | pin;
+
+	gpio_rz_pin_config_get_raw(port_pin, flags);
+	return 0;
+}
+#endif
+
+/* Get previous pin's configuration, used by pin_configure/pin_interrupt_configure api */
+static int gpio_rz_pin_config_get_raw(bsp_io_port_pin_t port_pin, uint32_t *flags)
+{
+	bsp_io_port_t port = (port_pin >> 8U) & 0xFF;
+	gpio_pin_t pin = port_pin & 0xFF;
+	volatile uint8_t *p_p = GPIO_RZ_IOPORT_P_REG_BASE_GET;
+	volatile uint16_t *p_pm = GPIO_RZ_IOPORT_PM_REG_BASE_GET;
+
+	uint8_t adr_offset;
+	uint8_t p_value;
+	uint16_t pm_value;
+
+	adr_offset = (uint8_t)GPIO_RZ_REG_OFFSET(port, pin);
+
+	p_p = &p_p[adr_offset];
+	p_pm = &p_pm[adr_offset];
+
+	p_value = GPIO_RZ_P_VALUE_GET(*p_p, pin);
+	pm_value = GPIO_RZ_PM_VALUE_GET(*p_pm, pin);
+
+	if (p_value) {
+		*flags |= GPIO_OUTPUT_INIT_HIGH;
+	} else {
+		*flags |= GPIO_OUTPUT_INIT_LOW;
+	}
+
+	*flags |= ((pm_value << 16));
+	return 0;
+}
+
+static int gpio_rz_pin_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_rz_config *config = dev->config;
+	struct gpio_rz_data *data = dev->data;
+	bsp_io_port_pin_t port_pin = config->fsp_port | pin;
+	uint32_t ioport_config_data = 0;
+	gpio_flags_t pre_flags;
+	fsp_err_t err;
+
+	gpio_rz_pin_config_get_raw(port_pin, &pre_flags);
+
+	if (!flags) {
+		/* Disconnect mode */
+		ioport_config_data = 0;
+	} else if (!(flags & GPIO_OPEN_DRAIN)) {
+		/* PM register */
+		if (flags & GPIO_INPUT) {
+			if (flags & GPIO_OUTPUT) {
+				ioport_config_data &= GPIO_RZ_PIN_CONFIGURE_INPUT_OUTPUT_RESET;
+				ioport_config_data |= IOPORT_CFG_PORT_DIRECTION_OUTPUT_INPUT;
+			} else {
+				ioport_config_data &= ~(0x3 << 2);
```

Comment:
Can we use a dedicated define here as well?

Reply:
Thank you for the feedback.
I changed to the dedicated defined macro.
Please kindly check it.